Modules/xSharePoint/DSCResources/MSFT_xSPUserProfileSyncConnection/MSFT_xSPUserProfileSyncConnection.psm1

```diff
@@ -0,0 +1,231 @@
+            $upcm = New-Object -TypeName Microsoft.Office.Server.UserProfiles.UserProfileConfigManager $context
+
+            $connection = $upcm.ConnectionManager | Where-Object { $_.DisplayName -eq $params.Name}
+            if($connection -eq $null){
+                return $null
+            }
+            $namingContext = $connection.NamingContexts | select -first 1
+            if($namingContext -eq $null){
+                return $null
+            }
+            $accountCredentials = "$($connection.AccountDomain)\$($connection.AccountUsername)"
+            $domainController = $namingContext.PreferredDomainControllers | select -First 1
+            return @{
+                        UserProfileService = $UserProfileService
+                        Forest = $connection.Server #"contoso.com" #TODO: GetCorrect Forest
+                        Name = $namingContext.DisplayName
+                        Credentials = $accountCredentials 
+                        IncludedOUs = $namingContext.ContainersIncluded
+                        ExcludedOUs = $namingContext.ContainersExcluded
+                        Server =$domainController
+                        UseSSL = $connection.UseSSL
+                        ConnectionType = $connection.Type.ToString()
+                        Force = $params.Force
+            }
+        }
+    }
+    return $result
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [parameter(Mandatory = $true)]  [System.String] $Name,
+        [parameter(Mandatory = $true)]  [System.String] $Forest,
+        [parameter(Mandatory = $true)]  [System.Management.Automation.PSCredential] $ConnectionCredentials,
+        [parameter(Mandatory = $true)] [System.String] $UserProfileService,
+        [parameter(Mandatory = $true)] [System.String[]] $IncludedOUs,
+        [parameter(Mandatory = $false)] [System.String[]] $ExcludedOUs,
+        [parameter(Mandatory = $false)] [System.String] $Server,
+        [parameter(Mandatory = $false)] [System.Boolean] $UseSSL,
+        [parameter(Mandatory = $false)] [System.Boolean] $Force,
+        [parameter(Mandatory = $false)] [System.String] $ConnectionType,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+   )
+
+    Write-Verbose -Message "Creating user profile service application $Name"
+
+
+    $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+    $params = $args[0]
+        
+
+    if ($params.ContainsKey("InstallAccount")) { $params.Remove("InstallAccount") | Out-Null }
+    $ups = Get-SPServiceApplication -Name $params.UserProfileService -ErrorAction SilentlyContinue 
+                
+    if ($null -eq $ups) { 
+        throw "User Profile Service Application $($params.UserProfileService) not found"
+    }
+    $context = Get-xSharePointServiceContext  $ups.ServiceApplicationProxyGroup 
```

Comment:
So, my problem was that to get the Context we need to have a url of an application. So for the ups property I got the CA url. I think this resource could/should use the same approach so I will update it before tuesday. shouldn't change much but I will have to update the unit tests.